command/repo.go

```diff
@@ -63,6 +80,146 @@ func repoClone(cmd *cobra.Command, args []string) error {
 	return utils.PrepareCmd(cloneCmd).Run()
 }
 
+func isURL(arg string) bool {
+	return strings.HasPrefix(arg, "http:/") || strings.HasPrefix(arg, "https:/")
+}
+
+func repoFork(cmd *cobra.Command, args []string) error {
+	ctx := contextForCommand(cmd)
+
+	forceYes, err := cmd.Flags().GetBool("yes")
+	if err != nil {
+		return err
+	}
+	forceNo, err := cmd.Flags().GetBool("no")
+	if err != nil {
+		return err
+	}
+
+	apiClient, err := apiClientForContext(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to create client: %w", err)
+	}
+
+	var toFork ghrepo.Interface
+	inParent := false // whether or not we're forking the repo we're currently "in"
+	if len(args) == 0 {
+		baseRepo, err := determineBaseRepo(cmd, ctx)
+		if err != nil {
+			return fmt.Errorf("unable to determine base repository: %w", err)
+		}
+		inParent = true
+		toFork = baseRepo
+	} else {
+		repoArg := args[0]
+
+		if isURL(repoArg) {
+			parsedURL, err := url.Parse(repoArg)
+			if err != nil {
+				return fmt.Errorf("did not understand argument: %w", err)
+			}
+
+			toFork, err = ghrepo.FromURL(parsedURL)
+			if err != nil {
+				return fmt.Errorf("did not understand argument: %w", err)
+			}
+
+		} else {
+			toFork = ghrepo.FromFullName(repoArg)
+			if toFork.RepoName() == "" || toFork.RepoOwner() == "" {
+				return fmt.Errorf("could not parse owner or repo name from %s", repoArg)
+			}
+		}
+	}
+
+	out := colorableOut(cmd)
+	fmt.Fprintf(out, "Forking %s...\n", utils.Cyan(ghrepo.FullName(toFork)))
+
+	authLogin, err := ctx.AuthLogin()
+	if err != nil {
+		return fmt.Errorf("could not determine current username: %w", err)
+	}
+
+	possibleFork := ghrepo.New(authLogin, toFork.RepoName())
+	exists, err := api.RepoExistsOnGitHub(apiClient, possibleFork)
```

Comment:
Ah, I just realized something. We do not have to check whether the fork already exists if you are going to fork otherwise. You can simply initiate the API fork operation and it will be a no-op and return the existing repo name if your fork already exists. That makes it simpler for you I think.

Reply:
that's a good point, i can play with that implementation.

Reply:
The create fork endpoint seems to behave exactly the same whether or not it created the repo; it returns 202 and the fork's information. This means I can't tell whether or not the fork already existed and we lose the ability to tell the user "hey, that thing you wanted to do is already done."

I personally like telling them that the fork already exists. Pretending that we created it (and offering to add a remote or clone) seems only like it will create confusion.

@ampinsk I'm curious what you think?

Reply:
It's a good point that we would prefer to communicate to the user what just happened.

One problem with looking up `YOURUSER/REPONAME` as possible fork repo is that a person _might_ have a fork of a repo, but that fork will not have the same repository name.

Example: I fork `vilmibm/dotfiles`. Because I already have a `dotfiles` repo, the fork will be created with the name https://github.com/mislav/dotfiles-1. Thus, we cannot assume that someone's fork will always have the same repository name.

Reply:
Agreed that I'd rather be transparent and clear about what's going on - so +1 on telling them the fork already exists in this scenario!

Reply:
I'm going to leave the repo-existence check in. That's a good point about the repo names; it seems like the right approach is to actually list the forks for the repo-to-fork and check for one belonging to the executing user. I'll take a stab at that.

Reply:
unfortunately there isn't an efficient way to do a more sophisticated check. the best i came up with was:
```
query { 
  viewer {
    repositories(isFork: true, first:100) {
      edges {
        node {
          name
          parent {
            owner {
              login
            }          }        }      }    }  }}
```

and then comparing the forks' login with the target repo's owner's login. I don't know that it's worth it at this point given that most of the time an existing fork is going to have the same name.

Reply:
There is an efficient way with GraphQL, at least in theory; in practice it looks like it's broken since it returns wrong results:
```graphql
{
  repository(owner:"github", name:"fetch") {
    forks(first: 1, affiliations: OWNER) {
      nodes {
        nameWithOwner
      }
    }
  }
}
```

I would advise against doing a lookup which we know in advance is imprecise. Right now the only precise API lookup we have is to kick off the fork operation and read the resulting repository. We could read the `created_at` attribute of the result to find out if the fork was just created so we can communicate that to the user.

Reply:
The `created_at` approach worked out pretty well. Once that gql query is fixed I'd rather switch to that.